src/ussd_nurse_rapidpro.js

```diff
@@ -370,7 +371,161 @@ go.app = function() {
+                    self.im.user.set_answer("dob", date_of_birth);
+                    }
+                },
+                next: 'state_end_detail_changed',
+            });
+        });
+
+    
+
+        self.states.add('state_passport', function(name) {
+            return new ChoiceState(name, {
+                question: $('What is the country of origin of the passport?'),
+                choices: [
+                    new Choice('na', $('Namibia')),
+                    new Choice('bw', $('Botswana')),
+                    new Choice('mz', $('Mozambique')),
+                    new Choice('sz', $('Swaziland')),
+                    new Choice('ls', $('Lesotho')),
+                    new Choice('cu', $('Cuba')),
+                    new Choice('other', $('Other')),
+                ],
+                next: 'state_passport_no'
+            });
+        });
+
+        self.states.add('state_passport_no', function(name) {
+            var error = $("Sorry, the format of the passport number is not correct. Please enter the passport number again.");
+            var question = $("Please enter the passport number:");
+            return new FreeText(name, {
+                question: question,
+                check: function(content) {
+                    if (!utils.is_alpha_numeric_only(content) || content.length <= 4) {
+                        return error;
+                    }
+                },
+                next: 'state_passport_dob'
+            });
+        });
+
+        self.states.add('state_passport_dob', function(name) {
+            var error = $("Sorry, the format of the date of birth is not correct. Please enter it again, e.g. 27 May 1975 as 27051975:");
+            var question = $("Please enter the date of birth, e.g. 27 May 1975 as 27051975:");
+            return new FreeText(name, {
+                question: question,
+                check: function(content) {
+                    if (!utils.is_valid_date(content, 'DDMMYYYY')) {
+                        return error;
+                    }
+                    else{
+                         /*
+                            add function to update id type info against operator id
+                            add function to update id number info against operator id
+                            add function to update date of birth info against operator id
+                            then lead to state where user details are successfully changed
+                        */
+                       var date_of_birth = moment(content, 'DDMMYYYY').format('YYYY-MM-DD');
+                        self.im.user.set_answer("id_type", "passport");
+                        self.im.user.set_answer("passport_no", self.im.user.answers.state_passport_no);
+                        self.im.user.set_answer("passport_origin", self.im.user.answers.state_passport);
+                        self.im.user.set_answer("dob", date_of_birth);
```

Comment:
Same here with these answers, we don't need to save them since we already have them automatically saved.